### PR TITLE
The "share" handling is improved

### PR DESCRIPTION
**Pull request by annando**
The code concerning the "share" functionality is simplified and unified.

Also There is a preparation for not using the "share" element anymore in the future when we share native posts.

### CODE REVIEW

File: src/Protocol/Diaspora.php [hunk 1]
```diff
@@ -3203,59 +3173,35 @@ public static function sendUnshare(array $owner, array $contact): int
 	/**
 	 * Checks a message body if it is a reshare
 	 *
-	 * @param string $body     The message body that is to be check
-	 * @param bool   $complete Should it be a complete check or a simple check?
+	 * @param array $item The message body that is to be check
 	 *
-	 * @return array|bool Reshare details or "false" if no reshare
+	 * @return array Reshare details or "false" if no reshare
```
Comment (MrPetovan) [inline src/Protocol/Diaspora.php]:
```suggestion
	 * @return array Reshare details (can be empty)
```

File: src/Protocol/Diaspora.php [hunk 2]
```diff
@@ -3203,59 +3173,35 @@ public static function sendUnshare(array $owner, array $contact): int
...
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 * @throws \ImagickException
 	 */
-	public static function isReshare(string $body, bool $complete = true)
+	public static function isReshare(array $item): array
```
Comment (MrPetovan) [inline src/Protocol/Diaspora.php]: This method should be renamed now that it always return an array.
```suggestion
	public static function getShareRootDetails(array $item): array
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Protocol/Diaspora.php (hunk 1 above)
Comment (annando) [reply src/Protocol/Diaspora.php]: Changed

File: src/Protocol/Diaspora.php (hunk 2 above)
Comment (annando) [reply src/Protocol/Diaspora.php]: Renamed

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**